solarforecastarbiter/reports/figures.py

```diff
@@ -0,0 +1,211 @@
+"""
+Functions to make all of the figures for Solar Forecast Arbiter reports.
+"""
+
+from bokeh.embed import components
+from bokeh.models import ColumnDataSource
+from bokeh.models.ranges import Range1d
+from bokeh.plotting import figure
+from bokeh import palettes
+
+import pandas as pd
+
+
+PALETTE = palettes.d3['Category10'][6]
+
+
+def format_variable_name(variable, units):
+    """Make a nice human readable name."""
+    caps = ('ac_', 'dc_', 'poa_', 'ghi', 'dni', 'dhi')
+    fname = variable
+    for cap in caps:
+        fname = fname.replace(cap, cap.upper())
+    fname = fname.replace('_', ' ')
+    return fname + f' ({units})'
+
+
+def line_or_step(interval_label):
+    if 'instant' in interval_label:
+        plot_method = 'line'
+        kwargs = dict()
+    elif interval_label == 'beginning':
+        plot_method = 'step'
+        kwargs = dict(mode='before')
+    elif interval_label == 'ending':
+        plot_method = 'step'
+        kwargs = dict(mode='after')
+    return plot_method, kwargs
+
+
+def construct_fx_obs_cds(fx_obs, fx_values, obs_values):
+    """
+    Construct a tuple of metadata, data for the plot functions.
+
+    Parameters
+    ----------
+    fx_obs : solarforecastarbiter.datamodel.ForecastObservation
+    fx_values : pandas.Series
+    obs_values : pandas.Series
+
+    Returns
+    -------
+    fx_obs : solarforecastarbiter.datamodel.ForecastObservation
+    cds : bokeh.models.ColumnDataSource
+    """
+    data = pd.DataFrame({'observation': obs_values, 'forecast': fx_values})
+    data['timestamp'] = data.index
+    cds = ColumnDataSource(data)
```

Comment:
does this care about the index? why not (rename and) reset_index and avoid the copied data?